modules/core/include/opencv2/core/mat.hpp

```diff
@@ -2364,31 +2364,34 @@ typedef Mat_<Vec3d> Mat3d;
 typedef Mat_<Vec4d> Mat4d;
 
 /** @todo document */
-class CV_EXPORTS UMat
+class CV_EXPORTS_W UMat
```

Comment:
This must be avoided.
We don't want to restrict C++ interface of this class and respect bindings restrictions. Like `cv::Mat` implement this using featutes of target language.

Reply:
@alalek Since `CV_EXPORTS_W` is an alias to `CV_EXPORTS`, C++ interface is not restricted whatsoever.

I cannot understand why you think this it needs to be revised, but if you insist, then there is actually no other way but to close the PR, since it fully depend on this exact line. What do you think?

Reply:
@alalek In fact, I believe that even `cv::Mat` needs to eventually be wrapped properly.
It is really a waste of processing power to convert the data to NumPy format back and forth before and after each function call. A better user control and performance gain can be achieved if the data stay in the C++ world (i.e. as `cv::Mat`) until the user explicitly require a conversion, similar to the current `UMat.get()` behavior.
You might think that Python is not about speed, but it is really becoming so. I hope you will consider your decision again..